SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
+			RequireEqual_Int(GetNumComponentsOnTestActor(), 4, TEXT("Now DynamicSubobjectTestActor should have 4 components"));
+			FinishStep();
+		},
+		StepTimeLimit);
+
+	for (int i = 0; i < DynamicComponentsPerClass + 2; ++i)
+	{
+		const bool LastStepLoop = i == DynamicComponentsPerClass + 1;
+
+		// Step 6 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
+		// AReplicatedGASTestActor.
+		AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation)
+				&& RequireEqual_Vector(ClientOneSpawnedPawn->GetActorLocation(), CharacterRemoteLocation,
+									   TEXT("Client pawn was not moved to remote location"), 1.0f))
+			{
+				FinishStep();
+			}
+		});
 
-	{ // Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+		// Step 7 - Client 1 makes sure that the movement was correctly replicated
 		AddStep(
-			TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+			TEXT("DynamicSubobjectsTestClientCheckFirstMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
 			[this](float DeltaTime) {
-				ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
-				APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
-				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
-				if (IsValid(PlayerController))
+				APawn* PlayerCharacter = GetFlowPawn();
+
+				if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should not be nullptr"))
+					&& RequireEqual_Vector(PlayerCharacter->GetActorLocation(), CharacterRemoteLocation,
+										   TEXT("Character was not moved to remote location"), 1.0f))
 				{
-					if (PlayerCharacter == PlayerController->AcknowledgedPawn)
-					{
-						FinishStep();
-					}
+					FinishStep();
 				}
 			},
 			StepTimeLimit);
-	}
 
-	for (int i = 0; i < DynamicComponentsPerClass + 1; ++i)
-	{
-		{ // Step 3 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
-		  // AReplicatedGASTestActor.
-			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
-				if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation))
+		// when in native, we need to wait for a while here - so the engine can update relevancy
+		USpatialNetDriver* CastNetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+		if (!CastNetDriver)
+		{
+			AddStep(TEXT("DynamicSubobjectsTestNativeWaitABit"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float DeltaTime) {
+				StepTimer += DeltaTime;
+				if (StepTimer > 6.0f)
 				{
-					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
-					{
-						FinishStep();
-					}
+					StepTimer = 0;
```

Comment:
I would prefer setting StepTimer = 0 in the one-shot step - make sure your preconditions are correct rather than setting up postconditions in hopes they will stay correct.

Reply:
done